doc/CMakeLists.txt

```diff
@@ -196,38 +207,30 @@ file(MAKE_DIRECTORY ${HTML_DIR})
 #-------------------------------------------------------------------------------
 # docset: Zephyr
 
-# Add the 'zephyr' target for building the Zephyr documentation. We reuse
-# doc/CMakeLists.txt from the Zephyr repository, but use our own Sphinx
-# configuration from doc/zephyr/conf.py. The generated HTML is placed in the
-# common Sphinx HTML output folder.
-
-# Parameters to doc/CMakeLists.txt in Zephyr. KCONFIG_OUTPUT is used to find
-# objects.inv from the Kconfig docs, to link the Zephyr docs to the Kconfig
-# docs.
-set(SPHINXOPTS ${SPHINXOPTS_DEFAULT} -c ${NRF_BASE}/doc/zephyr)
-set(SPHINX_OUTPUT_DIR ${HTML_DIR}/zephyr)
-set(KCONFIG_OUTPUT ${HTML_DIR}/kconfig)
-set(GEN_DEVICETREE_REST_ZEPHYR_DOCSET " ")
-
-# Get access to the 'html' target from doc/CMakeLists.txt in Zephyr
-set(MODULES_EXT_ROOT ${NRF_BASE})
-add_subdirectory(${ZEPHYR_BASE}/doc ${ZEPHYR_BINARY_DIR})
-
-add_custom_target(zephyr)
-add_dependencies(zephyr html)
-
-add_custom_target(zephyr-html)
-add_dependencies(zephyr-html html)
-
-add_custom_target(zephyr-html-all)
-add_dependencies(zephyr-html-all html)
+add_custom_target(
+  zephyr-devicetree
+  COMMAND ${CMAKE_COMMAND} -E env
+  PYTHONPATH=${ZEPHYR_BASE}/scripts/dts/python-devicetree/src
+  ZEPHYR_BASE=${ZEPHYR_BASE}
+  ${PYTHON_EXECUTABLE}
+    ${ZEPHYR_BASE}/doc/_scripts/gen_devicetree_rest.py
+    --vendor-prefixes ${ZEPHYR_BASE}/dts/bindings/vendor-prefixes.txt
+    --dts-root ${ZEPHYR_BASE}
+    ${ZEPHYR_BINARY_DIR}/src/reference/devicetree
+  VERBATIM
+  USES_TERMINAL
+)
 
-set_target_properties(
-  zephyr-html zephyr-html-all
-  PROPERTIES
-    ADDITIONAL_CLEAN_FILES "${ZEPHYR_BINARY_DIR};${SPHINX_OUTPUT_DIR}"
+set(zephyr_env
+  ZEPHYR_BASE=${ZEPHYR_BASE}
+  ZEPHYR_BUILD=${ZEPHYR_BINARY_DIR}
+  DOXYGEN_EXECUTABLE=${DOXYGEN_EXECUTABLE}
 )
 
+add_docset(zephyr "${zephyr_env}")
```

Comment:
This is the one actually adding all the targets right?

Reply:
yes, correct